solve : 사칙연산

ChUngHEe0227 opened the pull request:
문제 출처: 
사칙연산 : https://programmers.co.kr/learn/courses/30/lessons/1843
짝수행 세기 : https://programmers.co.kr/learn/courses/30/lessons/1843
도둑질 : https://programmers.co.kr/learn/courses/30/lessons/42897
입국 심사 :  https://programmers.co.kr/learn/courses/30/lessons/43238?language=python3
참고: 
사칙연산 :  https://real-012.tistory.com/62

ps: 어려웠음 시간 두고 천천히 봐 그냥 다음 문제로 이거 풀면서 참고로 코드 리뷰해도 될정도

ChUngHEe0227 committed:
solve : 사칙연산

문제 출처: https://programmers.co.kr/learn/courses/30/lessons/1843
참고: https://real-012.tistory.com/62

ChUngHEe0227 committed:
solve : 지형이동

문제 : https://programmers.co.kr/learn/courses/30/lessons/62050
-bfs : 같은 영역을 찾는 bfs
-find_height : 미니멈 사다리 위치를 찾는 함수

ByeongjaeJeon commented on CH_Python/Operations.py:
```diff
@@ -0,0 +1,38 @@
+        for j in range(0,len(arr)-i,2):    
+            tmax = 0
+            tmin = 0
+            max_n = -9999999
+            min_n = 9999999
```
어떤 기준으로 정한 숫자죠?

ChUngHEe0227 replied:
맥스넘 민넘

ByeongjaeJeon replied:
맥스는 1-1000까지 숫자가 2-101개 들어가니까 101000아닌가요

ChUngHEe0227 replied:
주석 처리 함

ByeongjaeJeon commented on CH_Python/Operations.py:
```diff
@@ -0,0 +1,38 @@
+            d[i][i][0] = oper[i]
+            d[i][i][1] = oper[i]
+
+    for i in range(2,len(arr),2): 
+        for j in range(0,len(arr)-i,2):    
```
i,j라니 못해도 3차원 배열이니 x,y,z 정도로라도 해주시죠

ChUngHEe0227 replied:
XYZ 랑 IJK 중 뭐가 나은지 는 의미 없는거 같에

ByeongjaeJeon replied:
풀어보니 그런 듯 그래도 보기 어려운 건 어려움ㅋㅋㅋ

ByeongjaeJeon reviewed (changes requested):
사칙 연산 코드를 읽어보려 했으나 전혀 눈에 들어오지 않아서 -1 일 때 +라 더하고 -2 일 때는 -라 빼고 대소 비교해서 tmax, tmin 넣는 거 말고는 뭐하는 건지 모르겠어요 선생님 
3차원 배열의 각 축이 뭔지 왜 그렇게 했는지 마지막에 tmax tmin 어디다 넣어서 어떻게 답 내는 건지 등등

ChUngHEe0227 committed:
solve : 짝수행 세기

푸는중

ChUngHEe0227 committed:
주석 추가

ChUngHEe0227 committed:
solve : 도둑질

문제 : https://programmers.co.kr/learn/courses/30/lessons/42897
-dp1 : 0번쨰 항을 이용하는 dp
-dp2 : 0번째 항을 이용하지 않는 dp

ChUngHEe0227 committed:
solve : 입국심사

- 시간을 이진 탐색 하는게 핵심
- 파이썬 이기 때문에 자료형은 무시할수 있는게 장점

ByeongjaeJeon reviewed (approved):
와! 드디어 approve!
푼 문제들은 저도 풀어볼게유 고생했으
